python/src/scipp/plot/profiler.py

```diff
@@ -0,0 +1,340 @@
+             ax=None,
+             cax=None,
+             aspect=None,
+             cmap=None,
+             log=False,
+             vmin=None,
+             vmax=None,
+             color=None,
+             logx=False,
+             logy=False,
+             logxy=False,
+             resolution=None):
+    """
+    Plot a 2D slice through a N dimensional dataset. For every dimension above
+    2, a slider is created to adjust the position of the slice in that
+    particular dimension.
+    """
+
+    sv = Profiler(scipp_obj_dict=scipp_obj_dict,
+                  axes=axes,
+                  masks=masks,
+                  ax=ax,
+                  cax=cax,
+                  aspect=aspect,
+                  cmap=cmap,
+                  log=log,
+                  vmin=vmin,
+                  vmax=vmax,
+                  color=color,
+                  logx=logx or logxy,
+                  logy=logy or logxy,
+                  resolution=resolution)
+
+    if ax is None:
+        render_plot(figure=sv.fig, widgets=sv.vbox, filename=filename)
+
+    return sv
+
+
+class Profiler(Slicer2d):
+    def __init__(self,
+                 scipp_obj_dict=None,
+                 axes=None,
+                 masks=None,
+                 ax=None,
+                 cax=None,
+                 aspect=None,
+                 cmap=None,
+                 log=None,
+                 vmin=None,
+                 vmax=None,
+                 color=None,
+                 logx=False,
+                 logy=False,
+                 resolution=None):
+
+        # Variables for profile plotting from pick need to be set before
+        # calling the init from Slicer2d as it calls update_axes() and
+        # update_slice(), which require these variables to be defined.
+        self.profile_viewer = None
+        self.profile_key = None
+        self.slice_pos_rectangle = None
+        self.profile_scatter = None
+        self.profile_update_lock = False
+        self.ax_pick = None
+        self.log = log
+
+        super().__init__(scipp_obj_dict=scipp_obj_dict,
+                         axes=axes,
+                         masks=masks,
+                         ax=ax,
+                         cax=cax,
+                         aspect=aspect,
+                         cmap=cmap,
+                         log=log,
+                         vmin=vmin,
+                         vmax=vmax,
+                         color=color,
+                         logx=logx,
+                         logy=logy,
+                         resolution=resolution)
+
+        # Initial checks
+        if self.ndim != 3:
+            raise RuntimeError("Picking on a 2d plot is only supported for 3D "
+                               "data.")
+
+        # Save a copy of the data array with bin-edge coords
+        self.da_with_edges = self.make_data_array_with_bin_edges()
+
+        # Double the figure height
+        self.fig.set_figheight(2 * self.fig.get_figheight())
```

Comment:
On my laptop I actually have trouble fitting this on a screen and have to resize the figure by hand. Is there anything better we can do, e.g., choose a more wide-screen (less high) format by default?

Reply:
Yes we can do more widescreen